src/Tasks/Microsoft.NET.Build.Tasks/CreateWindowsSdkKnownFrameworkReferences.cs

```diff
@@ -55,25 +55,34 @@ protected override void ExecuteCore()
             else
             {
                 var normalizedTargetFrameworkVersion = ProcessFrameworkReferences.NormalizeVersion(new Version(TargetFrameworkVersion));
+
+                var matchingKnownFrameworkReferences = new List<(Version minimumNetVersion, TaskItem knownFrameworkReference)>();
                 foreach (var supportedWindowsVersion in WindowsSdkSupportedTargetPlatformVersions)
                 {
                     var windowsSdkPackageVersion = supportedWindowsVersion.GetMetadata("WindowsSdkPackageVersion");
 
                     if (!string.IsNullOrEmpty(windowsSdkPackageVersion))
                     {
                         var minimumNETVersion = supportedWindowsVersion.GetMetadata("MinimumNETVersion");
+                        Version normalizedMinimumVersion = new Version(0, 0, 0);
                         if (!string.IsNullOrEmpty(minimumNETVersion))
                         {
-                            var normalizedMinimumVersion = ProcessFrameworkReferences.NormalizeVersion(new Version(minimumNETVersion));
+                            normalizedMinimumVersion = ProcessFrameworkReferences.NormalizeVersion(new Version(minimumNETVersion));
                             if (normalizedMinimumVersion > normalizedTargetFrameworkVersion)
                             {
                                 continue;
                             }
                         }
 
-                        knownFrameworkReferences.Add(CreateKnownFrameworkReference(windowsSdkPackageVersion, TargetFrameworkVersion, supportedWindowsVersion.ItemSpec));
+                        matchingKnownFrameworkReferences.Add((normalizedMinimumVersion, CreateKnownFrameworkReference(windowsSdkPackageVersion, TargetFrameworkVersion, supportedWindowsVersion.ItemSpec)));
+                        
                     }
                 }
+
+                //  If there are multiple matching WindowsSdkSupportedTargetPlatformVersion items, choose the one with the highest minimum version.  That way it is possible to
+                //  use older packages when targeting older versions of .NET, and newer packages for newer versions of .NET
+                var highestMinimumVersion = matchingKnownFrameworkReferences.Max(t => t.minimumNetVersion);
```

Comment:
I think you might need to also take into account the OS version too.  For instance, lets say we added a new OS TFM that is only supported in like .NET 7 and we added an minimumNetVersion entry for it. I think that would mean that would be the only one in the list when targeting .NET 7.

As an example, if you take a look at what we have right now, 10.0.22621.0 is defined for .NET 6 or higher.  I feel like with this change, only that would get selected and not the others when it should be the highest for the .NET version taking into account OS version.

https://github.com/dotnet/installer/blob/b3e1a28fb3579e3636c8918842cc5d6031ac3629/src/redist/targets/GenerateBundledVersions.targets#L874